rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -52,12 +70,14 @@ struct convert<rosbag2_storage::TopicMetadata>
     return node;
   }
 
-  static bool decode(const Node & node, rosbag2_storage::TopicMetadata & topic)
+  static bool decode(const Node & node, rosbag2_storage::TopicMetadata & topic, int version)
   {
     topic.name = node["name"].as<std::string>();
     topic.type = node["type"].as<std::string>();
     topic.serialization_format = node["serialization_format"].as<std::string>();
-
+    if (version >= 4) {
+      topic.offered_qos_profiles = node["offered_qos_profiles"].as<std::string>();
+    }
```

Comment:
do you want to reset `offered_qos_profiles` in an `else` branch to be on the safe side? Could `topic` contain a bogus value for this field when we start?

Reply:
it can't right now because it's a std::string default initialized immediately called before this function.